tools/Dockerfile

```diff
@@ -1,42 +1,46 @@
-FROM ubuntu:14.04
-
-RUN apt-get clean && \
-    apt-get update && \
-    apt-get install -y \
-        software-properties-common && \
-    apt-add-repository ppa:webupd8team/java && \
-    echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections && \
-    apt-get update && \
-    apt-get install -y \
-        oracle-java8-installer \
-        oracle-java8-set-default \
-        maven \
-        python2.7 \
-        curl \
-        build-essential \
-        libcurl4-openssl-dev \
-        git \
-        cmake \
-        libssl-dev \
-        uuid-dev \
-        valgrind \
-        libglib2.0-dev
-
-COPY build_nodejs.sh /tools/
-
-RUN /tools/build_nodejs.sh
-
-ENV JAVA_HOME /usr/lib/jvm/java-8-oracle
-
-ENV NODE_INCLUDE /build_nodejs/dist/inc
-ENV NODE_LIB /build_nodejs/dist/lib
-
-ENV GATEWAY_REPO https://github.com/Azure/azure-iot-gateway-sdk
-ENV COMMIT_ID master
-
-ENTRYPOINT rm -rf /gateway && \
-           git clone ${GATEWAY_REPO} /gateway && \
-           git -C /gateway checkout ${COMMIT_ID} && \
-           git -C /gateway submodule update --init && \
-           /gateway/tools/build.sh --run-unittests -rv --enable-java-binding --enable-nodejs-binding && \
-           bash
+FROM ubuntu:16.04
+
+RUN apt-get update && apt-get install -y apt-transport-https
+
+RUN sh -c 'echo "deb [arch=amd64] https://apt-mo.trafficmanager.net/repos/dotnet-release/ xenial main" > /etc/apt/sources.list.d/dotnetdev.list' \
+    && apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 417A0893 \
+    && apt-get update
+
+RUN apt-get install -y \
+    curl \
+    build-essential \
+    openjdk-8-jdk \
+    git \
+    cmake \
+    maven \
+    valgrind \
+    pkg-config \
+    uuid-dev \
+    libcurl4-openssl-dev \
+    libssl-dev \
+    libglib2.0-dev \
+    dotnet-dev-1.0.0-preview2.1-003177
+
+WORKDIR /
+
+RUN git clone https://github.com/Azure/azure-iot-gateway-sdk.git
+
+WORKDIR azure-iot-gateway-sdk/
+
+# Build node.dll for gateway's Node.js binding
+RUN cp tools/build_nodejs.sh . \
+    && ./build_nodejs.sh \
+    && rm ./build_nodejs.sh
+
+ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64 \
+    NODE_INCLUDE=/build_nodejs/dist/inc \
+    NODE_LIB=/build_nodejs/dist/lib
+
+# Trigger the population of the local package cache
+ENV NUGET_XMLDOC_MODE skip
+RUN mkdir /warmup \
+    && cd /warmup \
+    && dotnet new \
```

Comment:
@avranju While I have the info on my terminal screen, here's what happens when you run `dotnet new` for the first time:

```
Damons-MBP:~ damonb$ dotnet new console -o hwapp

Welcome to .NET Core!
---------------------
Learn more about .NET Core @ https://aka.ms/dotnet-docs. Use dotnet --help to see available commands or go to https://aka.ms/dotnet-cli-docs.

Telemetry
--------------
The .NET Core tools collect usage data in order to improve your experience. The data is anonymous and does not include command-line arguments. The data is collected by Microsoft and shared with the community.
You can opt out of telemetry by setting a DOTNET_CLI_TELEMETRY_OPTOUT environment variable to 1 using your favorite shell.
You can read more about .NET Core tools telemetry @ https://aka.ms/dotnet-cli-telemetry.

Configuring...
-------------------
A command is running to initially populate your local package cache, to improve restore speed and enable offline access. This command will take up to a minute to complete and will only happen once.
Decompressing 100% 4546 ms
Expanding 100% 5597 ms
Getting ready...
Content generation time: 49.3261 ms
The template "Console Application" created successfully.
Damons-MBP:~ damonb$ 
```

...so it looks like `dotnet new` does an _initial, system-wide_ population of your local package cache, while `dotnet restore` will do a project-specific population/refresh.